Grafana dashboards (#2494)
RedisLabs/redislabs-docs: content/rc/cloud-integrations/prometheus-integration.md

---
Title: Prometheus integration with Redis Cloud
linkTitle: Prometheus integration
description: Use Prometheus and Grafana to collect and visualize Redis Cloud metrics.
weight: $weight
alwaysopen: false
categories: ["RC"]
---

You can use Prometheus and Grafana to collect and visualize your Redis Enterprise Cloud metrics.

- [Prometheus](https://prometheus.io/) is an open source systems monitoring and alerting toolkit that can scrape metrics from different sources.
- [Grafana](https://grafana.com/) is an open source metrics visualization tool that can process Prometheus data.

Redis Cloud exposes its metrics through a Prometheus endpoint. You can configure your Prometheus server to scrape metrics from your Redis Cloud subscription on port 8070.

> The Redis Cloud Prometheus endpoint is exposed on Redis Cloud's internal network. To access this network, you need to enable [VPC peering](({{< relref "/rc/security/vpc-peering" >}})). VPC peering is only available with Flexible or Annual subscriptions. Because VPC peering is not available on Fixed or Free subscriptions, you cannot use Prometheus and Grafana with Fixed or Free subscriptions.

Reviewer: @cmilesb @banker we now also support node exporter access over Private Service Connect and within a couple of months also over Transit Gateway. I wonder whether we should remove the part about VPC peering and instead just say that this is available for Flexible/Annual subscriptions only.

Reviewer 2: ```suggestion
The Redis Cloud Prometheus endpoint is exposed on Redis Cloud's internal network. To access this network, you need to enable [VPC peering]({{< relref "/rc/security/vpc-peering" >}}) or [Private Service Connect]({{< relref "/rc/security/private-service-connect" >}}). Both options are only available with Flexible or Annual subscriptions. You cannot use Prometheus and Grafana with Fixed or Free subscriptions.
```

@joniredis Would this work?

> The Redis Cloud Prometheus endpoint is exposed on Redis Cloud's internal network. To access this network, you need to enable [VPC peering]({{< relref "/rc/security/vpc-peering" >}}) or [Private Service Connect]({{< relref "/rc/security/private-service-connect" >}}). Both options are only available with Flexible or Annual subscriptions. You cannot use Prometheus and Grafana with Fixed or Free subscriptions.

Reviewer 3: ```suggestion
The Redis Cloud Prometheus endpoint is exposed on Redis Cloud's internal network. To access this network, enable [VPC peering]({{< relref "/rc/security/vpc-peering" >}}) or [Private Service Connect]({{< relref "/rc/security/private-service-connect" >}}). Both options are only available with Flexible or Annual subscriptions. You cannot use Prometheus and Grafana with Fixed or Free subscriptions.
```

    internal.<cluster_address>:8070
    ``` 

> 1. Create an instance to run Prometheus and Grafana on the same cloud provider as your Redis Cloud subscription (for example, Amazon Web Services or Google Cloud Platform). This instance must:

Reviewer: @cmilesb Google changes the name of Google Cloud Platform to just "Google Cloud" so any reference to GCP should be changed to the latter.

> 1. Create an instance to run Prometheus and Grafana on the same cloud provider as your Redis Cloud subscription (for example, Amazon Web Services or Google Cloud Platform). This instance must:

Reviewer 2: ```suggestion
1. Create an instance to run Prometheus and Grafana on the same cloud provider as your Redis Cloud subscription (for example, Amazon Web Services or Google Cloud). This instance must:
```

    {{< /note >}}

1. Add dashboards for your subscription and database metrics.
    To add preconfigured dashboards:
    1. In the Grafana dashboards menu, select **Manage**.
    1. Select **Import**.
    1. Add the [subscription status](https://grafana.com/grafana/dashboards/18406-subscription-status-dashboard/) and [database status](https://grafana.com/grafana/dashboards/18407-database-status-dashboard/) dashboards.

### Grafana dashboards for Redis Cloud

> We publish two preconfigured dashboards for Redis Cloud and Grafana:

Reviewer 3: ```suggestion
Redis Enteprise publishes two preconfigured dashboards for Redis Cloud and Grafana:
```

> We publish two preconfigured dashboards for Redis Cloud and Grafana:

Reviewer 2: ```suggestion
Redis publishes two preconfigured dashboards for Redis Cloud and Grafana:
```

* The [subscription status dashboard](https://grafana.com/grafana/dashboards/18406-subscription-status-dashboard/) provides an overview of your Redis Cloud subscriptions.

> * The [database status dashboard](https://grafana.com/grafana/dashboards/18408-database-status-dashboard/) displays specific database metrics, including latency, memory usage, ops/second, key count.

Reviewer 3: ```suggestion
* The [database status dashboard](https://grafana.com/grafana/dashboards/18408-database-status-dashboard/) displays specific database metrics, including latency, memory usage, ops/second, and key count.
```

> These dashboards are open source. For additional dashboard options, or to file an issue, see the [Redis Enterprise Grafana Dashboards Github repository(https://github.com/redis-field-engineering/redis-enterprise-grafana-dashboards).

Reviewer 2: ```suggestion
These dashboards are open source. For additional dashboard options, or to file an issue, see the [Redis Enterprise Grafana Dashboards Github repository](https://github.com/redis-field-engineering/redis-enterprise-grafana-dashboards).
```

> For more information about configuring Grafana dashboards in general, see the [Grafana documentation](https://grafana.com/docs/).

Reviewer 3: ```suggestion
For more information about configuring Grafana dashboards, see the [Grafana documentation](https://grafana.com/docs/).
```
